os/include/tinyara/pm/pm.h

```diff
@@ -350,9 +363,42 @@ struct pm_callback_s {
 	 *
 	 **************************************************************************/
 
+#ifdef  CONFIG_IDLE_PM
+	void (*notify)(FAR struct pm_callback_s *cb, int domain, int pmstate);
+#else
 	void (*notify)(FAR struct pm_callback_s *cb, int domain, enum pm_state_e pmstate);
+#endif
+};
+
+#ifdef  CONFIG_IDLE_PM
+
+/* This structure provides the tick count threshold to enter all the  power
+ * management states that can be supported by the SOC & pointers to the
+ * functions supported by the SOC for changing to those states. These return
+ * OK on successful execution and -1 on any error encountered.
+ */
+
+/* Number of sleep states supported by SOC */
+
+int sleep_states_count;
+
+struct pm_idle_state_info {
+
+	/* Function pointers to state change function calls */
+
+	void (*state_fptr)(void);
```

Comment:
since it is a generic framework, it is better accept return value to handle error conditions.